### PR TITLE
Add integration tests for anonymous auth

### PR DESCRIPTION
**Pull request by samhorlbeck**

### CODE REVIEW

File: packages-exp/auth-exp/test/integration/flows/anonymous.test.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    it('account can be upgraded by setting email and password', async () => {
+      const { user } = await signInAnonymously(auth);
+      await updateEmail(user, email);
+      await updatePassword(user, 'password');
+
+      const anonId = user.uid;
+      await auth.signOut();
+      expect(
+        (await signInWithEmailAndPassword(auth, email, 'password')).user.uid
```
Comment (yuchenshi) [inline packages-exp/auth-exp/test/integration/flows/anonymous.test.ts:95]: nit: please split this out into its own variable.

File: packages-exp/auth-exp/test/integration/flows/anonymous.test.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      ).to.eq(anonId);
+    });
+
+    it('account can be linked using email and password', async () => {
+      const { user } = await signInAnonymously(auth);
```
Comment (yuchenshi) [inline packages-exp/auth-exp/test/integration/flows/anonymous.test.ts:100]: I'd suggest renaming the variable to `anonUser` (like below) to make the test more readable. You can also drop the `id` constant -- it's not much longer but clearer to read in its full form.

File: packages-exp/auth-exp/test/integration/flows/anonymous.test.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  const auth = initIntegrationTestContext();
+
+  it('signs in anonymously', async () => {
+    const userCred = await signInAnonymously(auth);
+    expect(auth.currentUser).to.eq(userCred.user);
+    expect(userCred.operationType).to.eq(OperationType.SIGN_IN);
+
+    const user = userCred.user;
+    expect(user.isAnonymous).to.be.true;
+    expect(typeof user.uid).to.eq('string');
```
[the remark this answers is not in the document]
Comment (samhorlbeck) [reply packages-exp/auth-exp/test/integration/flows/anonymous.test.ts]: Done

File: packages-exp/auth-exp/test/integration/flows/anonymous.test.ts (hunk 1 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/test/integration/flows/anonymous.test.ts]: Done

File: packages-exp/auth-exp/test/integration/flows/anonymous.test.ts (hunk 2 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/test/integration/flows/anonymous.test.ts]: Done

**Pull request merged**